### PR TITLE
auto update rep score, use rep score for cosoul art

### PR DESCRIPTION
**Pull request by CryptoGraffe**
event triggers update all the scores automatically now 

cosoul is generated using repscore as long as it is higher than pGIVE

### <samp>🤖 Generated by Copilot at 0ac10d4</samp>

This pull request implements the reputation score feature for the co-souls. The reputation score is calculated from the poap data and the co-soul activities, such as linking, creating, and updating co-souls and profiles. The reputation score is stored in the database and updated by event triggers. The reputation score is also displayed on the co-soul art components. The feature is controlled by a feature flag. The pull request also includes some code refactoring and readability improvements.

### <samp>🤖 Generated by Copilot at 0ac10d4</samp>

> _We're updating the rep score of the co-souls_
> _With event triggers and webhooks and poaps_
> _We're adding the props and the flags and the modules_
> _Heave away, me hearties, heave away_

### <samp>🤖 Generated by Copilot at 0ac10d4</samp>

*  Add event triggers and handler functions to update the reputation score of the co-soul address, the link holder, and the profile based on various actions ([link](https://github.com/coordinape/coordinape/pull/2415/files?diff=unified&w=0#diff-b4b6da422370f53365da30ca3a67cb779e6bb397e7df6b6b065ef4e55ac1c5b5R1-R27), [link](https://github.com/coordinape/coordinape/pull/2415/files?diff=unified&w=0#diff-3b9e4808597475aa9d3086b932c3f046df1e08a785e9d81ba0a6b596488d8e33R1-R24), [link](https://github.com/coordinape/coordinape/pull/2415/files?diff=unified&w=0#diff-f4450016ddacb5ef9d95ab739174229dc9560b68c6b56a66322d65339fa0e4ebR1-R22), [link](https://github.com/coordinape/coordinape/pull/2415/files?diff=unified&w=0#diff-1ff9bd45a785b1be61a7dc4d3b900604ffeaaf8c11f49e2c81fcfdff758d10f0R25-R27), [link](https://github.com/coordinape/coordinape/pull/2415/files?diff=unified&w=0#diff-1ff9bd45a785b1be61a7dc4d3b900604ffeaaf8c11f49e2c81fcfdff758d10f0R66-R71), [link](https://github.com/coordinape/coordinape/pull/2415/files?diff=unified&w=0#diff-3837acf731f554c94063419d1b6e3b165007c982ac40c9d93b10416b78fd9532R82-R99), [link](https://github.com/coordinape/coordinape/pull/2415/files?diff=unified&w=0#diff-e3fcc720cfb4d3d9d9f3784aea437911ecc54128562dc0d665da2d734348a11aR19-R36), [link](https://github.com/coordinape/coordinape/pull/2415/files?diff=unified&w=0#diff-b169d0a124485038732434cebea3338751b526a509d138dc344478947734b515R27-R44), [link](https://github.com/coordinape/coordinape/pull/2415/files?diff=unified&w=0#diff-10e1f74e3d2d69ec746a7e953c84d7c2466c1125e2893476dcbc582b29364e73R44-R61), [link](https://github.com/coordinape/coordinape/pull/2415/files?diff=unified&w=0#diff-5d6ef7e66a632964d24b9f0b42793ee3f15f5c51ba4add0a9d5385febb34e9a2R22-R39), [link](https://github.com/coordinape/coordinape/pull/2415/files?diff=unified&w=0#diff-c4ced3a654ee75b7c1ea1106d4d957286b19bb0bd93d7fcd9983343fb788529bR36-R53))
*  Modify the `syncPoapDataForAddress` function in `poap-api.ts` to call the `updateRepScoreForAddress` function after fetching the poap data ([link](https://github.com/coordinape/coordinape/pull/2415/files?diff=unified&w=0#diff-1c94cb78d152557eb07f7b63c6a870a6358304fd2f3d0255335c2781f7804761R216-R218))
*  Add the `repScore` field to the return type, query, and response of the `getCosoulData` function in `cosoul/[address].ts` ([link](https://github.com/coordinape/coordinape/pull/2415/files?diff=unified&w=0#diff-c05b4221a8fa5c2d52523c14cdf221c5dc469b8ee2f4a6dd47a3cc32babf6852R66), [link](https://github.com/coordinape/coordinape/pull/2415/files?diff=unified&w=0#diff-c05b4221a8fa5c2d52523c14cdf221c5dc469b8ee2f4a6dd47a3cc32babf6852R79-R88), [link](https://github.com/coordinape/coordinape/pull/2415/files?diff=unified&w=0#diff-c05b4221a8fa5c2d52523c14cdf221c5dc469b8ee2f4a6dd47a3cc32babf6852R254))
*  Add the `repScore` field to the query and response of the `getCosoulArtData` function in `cosoul/art/[artTokenId].ts` ([link](https://github.com/coordinape/coordinape/pull/2415/files?diff=unified&w=0#diff-8a513163f8449f612956f9c609b9b079493b1c71f86cd13dc2f8d3902724387aR41-R45), [link](https://github.com/coordinape/coordinape/pull/2415/files?diff=unified&w=0#diff-8a513163f8449f612956f9c609b9b079493b1c71f86cd13dc2f8d3902724387aR63))
*  Add the `repScore` prop to the `CoSoulArt` component in `MintPage.tsx` and `ViewPage.tsx` ([link](https://github.com/coordinape/coordinape/pull/2415/files?diff=unified&w=0#diff-a9d33a03447e2b96e70bddf1ebde47791709f7845cfffb8c05a3ee3dacf0348fL72-R76), [link](https://github.com/coordinape/coordinape/pull/2415/files?diff=unified&w=0#diff-07a5e546156e72d3af4da968af148298858d1943242859c44c28aea5e4b3c5b5L89-R93))
*  Add a call to the `updateRepScore` function in `login.ts` to update the reputation score of the inviter when a new user is invited ([link](https://github.com/coordinape/coordinape/pull/2415/files?diff=unified&w=0#diff-01abf081aaf21c9dd7749fceb2eaf3c000f88207e51f4a1ab234455045e4d2deR21), [link](https://github.com/coordinape/coordinape/pull/2415/files?diff=unified&w=0#diff-01abf081aaf21c9dd7749fceb2eaf3c000f88207e51f4a1ab234455045e4d2deR198-R202))
*  Add a comment to the `syncCoSouls` function in `syncCoSouls.ts` to indicate that the repScore should be updated on chain if needed ([link](https://github.com/coordinape/coordinape/pull/2415/files?diff=unified&w=0#diff-bec200d548388ad9691f929432061db782d0630390749744eaabb7a4c71928ebR74-R76))
*  Change the import order of the `poap-api.ts` file to follow the alphabetical order of the modules ([link](https://github.com/coordinape/coordinape/pull/2415/files?diff=unified&w=0#diff-1c94cb78d152557eb07f7b63c6a870a6358304fd2f3d0255335c2781f7804761L3-R12))
*  Remove the eslint-disable comment from the `muteChanged.ts` file ([link](https://github.com/coordinape/coordinape/pull/2415/files?diff=unified&w=0#diff-96b4b7351a08d5bb4fa0459decec8564f16b9767a315141401facec58761862bL1))
*  Add an empty line to the `login.ts` file to separate the code blocks for readability ([link](https://github.com/coordinape/coordinape/pull/2415/files?diff=unified&w=0#diff-01abf081aaf21c9dd7749fceb2eaf3c000f88207e51f4a1ab234455045e4d2deR237))
*  Remove an empty line from the `updateHolders.ts` file to remove unnecessary whitespace ([link](https://github.com/coordinape/coordinape/pull/2415/files?diff=unified&w=0#diff-5c06f186f3807378049f50192f6f66d97e0e57e242104d8d587c4591950b2e79L254))
*  Add a comment to the `deleteFromLinkHolderCacheAndPrivateVisibility` function in `updateHolders.ts` to explain the logic of deleting the visibility ([link](https://github.com/coordinape/coordinape/pull/2415/files?diff=unified&w=0#diff-5c06f186f3807378049f50192f6f66d97e0e57e242104d8d587c4591950b2e79R421))
*  Add the `rep_cosouls` feature name and flag to the `features.ts` file ([link](https://github.com/coordinape/coordinape/pull/2415/files?diff=unified&w=0#diff-5c118d9192f4171e7f6592b2205521832975d0666a8673815188b73c9644088aR10), [link](https://github.com/coordinape/coordinape/pull/2415/files?diff=unified&w=0#diff-5c118d9192f4171e7f6592b2205521832975d0666a8673815188b73c9644088aL15-R18))

### CODE REVIEW

File: api-lib/event_triggers/muteChanged.ts [hunk 1]
```diff
@@ -1,4 +1,3 @@
-/* eslint-disable no-console */
```
Comment (crabsinger) [inline api-lib/event_triggers/muteChanged.ts:1]: remove line 28 with the console.log

File: api-lib/event_triggers/updateProfileRepScore.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import type { VercelRequest, VercelResponse } from '@vercel/node';
+
+import { updateRepScore } from '../../src/features/rep/api/updateRepScore';
+import { EventTriggerPayload } from '../types';
+
+export default async function handler(req: VercelRequest, res: VercelResponse) {
+  try {
+    const payload: EventTriggerPayload<
+      'twitter_accounts' | 'github_accounts' | 'linkedin_accounts' | 'emails',
```
Comment (crabsinger) [inline api-lib/event_triggers/updateProfileRepScore.ts:9]: pull the table list out for easier reading and adding

File: api/hasura/cron/syncCoSouls.ts [hunk 3]
```diff
@@ -71,6 +71,9 @@ export async function syncCoSouls() {
         'onChainPGIVE:',
         onChainPGIVE
       );
+
+      // update repScore on chain if needed
```
Comment (crabsinger) [inline api/hasura/cron/syncCoSouls.ts:75]: is this a TODO? @CryptoGraffe

File: api-lib/event_triggers/updateProfileRepScore.ts (hunk 2 above)
Comment (crabsinger) [reply api-lib/event_triggers/updateProfileRepScore.ts]: i fixed this in cleanup pr

File: api-lib/event_triggers/muteChanged.ts (hunk 1 above)
Comment (crabsinger) [reply api-lib/event_triggers/muteChanged.ts:1]: i fixed in cleanup pr

**Review by crabsinger — APPROVED** [review]
code looks good - one question on syncCoSouls

Do we need the "Update Score" button for any reasno now?
Screenshot 2023-11-10 at 9 11 39 AM

@CryptoGraffe

File: api/hasura/cron/syncCoSouls.ts (hunk 3 above)
Comment (CryptoGraffe) [reply api/hasura/cron/syncCoSouls.ts:75]: diff PR

**Pull request merged**